automation_tools/__init__.py

```diff
@@ -1767,6 +1767,23 @@ def setenforce(mode):
 # =============================================================================
 # Satellite and Capsule Upgrade
 # =============================================================================
+
+def reboot(host, halt_time=300):
```

Comment:
You can refactor the reboot to be a generic task and not require the host... the host can be defined from outside using execute.

```python
def reboot(halt_time=300):
    """Reboots the the given host.

    Also halts the execution untill reboots according to given time.

    :param string host: Hostname of host to reboot.
    :param int halt_time: Halt execution in seconds.
    """
    print('Rebooting the host, please wait .... '.format(host))
    try:
        run('reboot', warn_only=True)
    except:
        pass
    time.sleep(halt_time)

def other_task():
    execute(reboot, host=host)
```